pandas/tests/reshape/test_cut.py

```diff
@@ -624,4 +624,21 @@ def test_cut_nullable_integer(bins, right, include_lowest):
         pd.array(a, dtype="Int64"), bins, right=right, include_lowest=include_lowest
     )
     expected = cut(a, bins, right=right, include_lowest=include_lowest)
+
+    tm.assert_categorical_equal(result, expected)
+
+
+@pytest.mark.parametrize("bins", [2, [0, 50, 100]])
+@pytest.mark.parametrize("right", [True, False])
+@pytest.mark.parametrize("include_lowest", [True, False])
+def test_cut_object_dtype_with_na(bins, right, include_lowest):
```

Comment:
can u add a test for an object array that won’t coerce (eg raises)

also for a string array (again should raise)

the key is a useful error message

Reply:
Looks like this is a bit harder than I thought to fix in general (having some trouble figuring out how to cast _some_ but not _all_ object arrays that contain number-like values; being as liberal as I am here causes `value_counts` tests to fail for instance).

Since the original issue actually appears to be caused by a `searchsorted` bug in numpy that is relatively hard to reproduce (https://github.com/numpy/numpy/issues/15499), do you think it makes sense to leave as is and wait for it to be fixed upstream?